### PR TITLE
Fixes `typesof()` for non-paths

### PR DESCRIPTION
**Pull request by ike709**
Replicates BYOND behavior

### CODE REVIEW

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs [hunk 1]
```diff
@@ -1854,7 +1854,9 @@ public static DreamValue NativeProc_splittext(DreamObject instance, DreamObject
             DreamList list = DreamList.Create();
 
             foreach (DreamValue type in arguments.GetAllArguments()) {
-                DreamPath typePath = type.GetValueAsPath();
+                if (!type.TryGetValueAsPath(out var typePath)) {
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:1857]: This proc also accepts DreamObjects

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs (hunk 1 above)
Comment (wixoaGit) [reply OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:1857]: This proc apparently also takes strings like `"/obj"`. DreamLists need to be ignored too.

**Comment by ike709** [comment]
It's not as optimal as possible but it should work.

**Comment by out-of-phaze** [comment]
It doesn't work for things like `typesof("/datum/proc")`. Which works in DM, for some absurd reason! Gives a list of all procs defined on datum.

**Comment by out-of-phaze** [comment]
> Separate issue, out of scope :^)

I literally gave you a patch that solves it

**Comment by ike709** [comment]
I commented that _before_ you did :^)

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**